### PR TITLE
:bug: Use natural sort order

### PR DESCRIPTION
**Pull request by rszwajko**
Replace default alphabetical sort order (i.e. a, a11, a2)               
with human friendly "natural" sort (i.e. a, a2, a11).                   
The change applies to all found cases of string sorting.                
For convenience the comparator converts all values to strings (with     
nulish values being converted to empty string).                         
                                                                        
In addition, the natural sort with string conversion (universal          
comparator) was used as a replacement in sorting hooks. There a general 
sorting algorithm was implemented with type-specific comparators.     
Benefits of universal comparator:                                       
1. better handling of mixed types (i.e. string and number). String value
   most likely gets converted to NaN.                                   
2. better handling of undefined (numeric value NaN)                     
2. simplified coercion logic compared to less than/greater then         
   operators 

Note that one functional change was introduced in the sorting hook:
space removal logic was dropped as it creates visually confusing results 
for stable sorting.

Resolves: #1796 

Reference-Url: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Less_than

### CODE REVIEW

**Review by ibolton336 — APPROVED** [review]

File: client/src/app/hooks/table-controls/sorting/getLocalSortDerivedState.ts [hunk 1]
```diff
@@ -51,22 +52,10 @@ export const getLocalSortDerivedState = <
...
-      return activeSort.direction === "asc" ? aValue - bValue : bValue - aValue;
-    } else {
-      if (aValue > bValue) return activeSort.direction === "asc" ? -1 : 1;
-      if (aValue < bValue) return activeSort.direction === "asc" ? -1 : 1;
-    }
-
-    return 0;
+    const aValue = getSortValues(a)[activeSort.columnKey];
+    const bValue = getSortValues(b)[activeSort.columnKey];
+    const compareValue = universalComparator(aValue, bValue, i18n.language);
```
Comment (sjd78) [inline client/src/app/hooks/table-controls/sorting/getLocalSortDerivedState.ts:57]: Looks like this is missing the space collapsing that was there before. Does that change any thing?

**Review by sjd78 — COMMENTED** [review]
Look really good, just a handful of questions

File: client/src/app/hooks/useLegacySortState.ts [hunk 2]
```diff
@@ -49,27 +50,10 @@ export const useLegacySortState = <T>(
...
-        else return bValue - aValue;
-      } else {
-        if (aValue > bValue) return direction === "asc" ? -1 : 1;
-        if (aValue < bValue) return direction === "asc" ? -1 : 1;
-      }
-
-      return 0;
+      const aValue = getSortValues(a)[index || 0];
+      const bValue = getSortValues(b)[index || 0];
+      const compareValue = universalComparator(aValue, bValue, i18n.language);
```
[the remark this answers is not in the document]
Comment (rszwajko) [reply client/src/app/hooks/useLegacySortState.ts:55]: Right. I should probably mention it as one of the changes. 
First, it's space removal not space collapsing. Therefore, IMHO it creates visually confusing results for stable sorting i.e.
```
const arr = ["a part", "apart", "a part"];
arr.sort((a,b) => { 
   const aValue = a.replace(/ +/g, "");
   const bValue = b.replace(/ +/g, "");
   return aValue.localeCompare(bValue, "en");
} );
# result: 
# ["a part", "apart", "a part"]

[the remark this answers is not in the document]
Comment (sjd78) [reply client/src/app/hooks/useLegacySortState.ts:55]: Yeah space removal isn't good.

Space collapsing doesn't seem to be needed either:
```js
const s1 = (a, b) => a.localeCompare(b, "en", { numeric: true });

const s2 = (a, b) => a.replace(/\s+/g, " ").localeCompare(b.replace(/\s+/g, " "), "en", { numeric: true });

["apart", "a   part", "", "a        part", "a parts", ""].sort()
// output: [ '', '', 'a        part', 'a   part', 'a parts', 'apart' ]

["apart", "a   part", "", "a        part", "a parts", ""].sort(s1)
// output: [ '', '', 'a        part', 'a   part', 'a parts', 'apart' ]

["apart", "a   part", "", "a        part", "a parts", ""].sort(s2)
// output: [ '', '', 'a   part', 'a        part', 'a parts', 'apart' ]

```

[the remark this answers is not in the document]
Comment (rszwajko) [reply client/src/app/hooks/useLegacySortState.ts:55]: I've updated the PR description to notify about this functional change.

**Review by sjd78 — APPROVED** [review]
LGTM

**Pull request merged**